doc/nrf/gs_modifying.rst

```diff
@@ -95,15 +122,80 @@ Typically, the :file:`CMakeLists.txt` files include at least the :file:`main.c`
+   In most cases this will be your application :file:`prj.conf` file.
 
 For detailed information about configuration options, see :ref:`zephyr:application-kconfig` in the Zephyr documentation.
 
+Hardware-related configuration
+==============================
+
+The hardware-related configuration comes from :file:`dts` files.
+There are several layers that provide the hardware definitions:
+
+* The lowest layer consists of defaults provided in :file:`dtsi` files for the given SoC.
+* The :file:`dtsi` files are included by selected board in its :file:`dts` file, which provides defaults of hardware configuration for the entire board.
+* The hardware-related configuration is further overlayed by the :file:`app.overlay` file, which is provided by the application.
+
+After merging the configuration sets imported from these configuration files, the build system creates a single :file:`zephyr.dts` file that describes the entire hardware-related configuration of the system.
+This process also creates a set of header files that provide defines, which can be used to refer hardware configuration from the source code.
+The hardware-related configuration does not change until the project is reevaluated.
+
+.. note::
+   The files resulting from DTS evaluation should not be notified.
+   To store the configuration parmanantly, you need to identify the file that contributes to the specific part of DTS tree and edit it directly.
+   In most cases, this is the :file:`app.overlay` file of your application.
+
+Partition manager configuration
+===============================
+
+If enabled, the :ref:`partition_manager` provides the memory layout definition.
+The layout is impacted by various elements, such as Kconfig configuration options or presence of child images.
+The build system works to ensure that all required partitions are enabled and that the right size of each partition area is selected.
+If you want to provide a custom partition definition, you can do this by specifying the static configuration in the :file:`pm_static.yml` file.
+
+After the project evaluation, the build system creates a single :file:`partitions.yml` file that describes the resulting memory layout.
+This process also creates a set of header files that provides defines, which can be used to refer to memory layout elements.
+The memory layout does not change until the project is reevaluated.
+
+Child images
+============
+
+The |NCS| build system allows the application project to become a root for sub-applications referred in the |NCS| as :ref:`child images <ug_multi_image>`.
+The child images are system components that reside on the same hardware system.
+They either cooperate with the application or the application depends on them.
+An example of child images are bootloaders, network cores, or security-related binaries.
+
+Each child image is a separate application.
+However, when constructed as part of your application, each child image configuration can be controlled by it.
```

Comment:
What does this mean ?

Does the parent image control the complete build of the child image, or can the user configure all or part of the child image build ?

Reply:
That would depend on how you look. I would say it controls child image build as a whole.

Reply:
I would remove this sentence and instead direct the user to the multi image page.

Something like:
```suggestion
Configuration of child images are described in the [Defining and enabling a child image](http://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/ug_multi_image.html#defining-and-enabling-a-child-image) section.
```

Reply:
We can do that but I would like to have documentation informing the user how build type propagates. Did not want the child images jump out of the box on this page so tried to introduce them. Can do that with a link and we will see how it looks.

Reply:
Added the new sentence. Removed the "However" one.

Reply:
> Did not want the child images jump out of the box on this page so tried to introduce them. 

and you do, which I think is fine and i'm not asking you to remove the subsection, but I don't think we should go into details here, instead we should direct the reader to the relevant page.